oap-server/server-receiver-plugin/skywalking-trace-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/trace/provider/parser/listener/service/ServiceMappingSpanListener.java

```diff
@@ -58,14 +71,20 @@ private ServiceMappingSpanListener(ModuleManager moduleManager) {
         if (!spanDecorator.getSpanLayer().equals(SpanLayer.MQ)) {
             if (spanDecorator.getRefsCount() > 0) {
                 for (int i = 0; i < spanDecorator.getRefsCount(); i++) {
-                    int serviceId = serviceInventoryCache.getServiceId(spanDecorator.getRefs(i).getNetworkAddressId());
-                    int mappingServiceId = serviceInventoryCache.get(serviceId).getMappingServiceId();
-                    if (mappingServiceId != segmentCoreInfo.getServiceId()) {
-                        ServiceMapping serviceMapping = new ServiceMapping();
-                        serviceMapping.setServiceId(serviceId);
+                    int networkAddressId = spanDecorator.getRefs(i).getNetworkAddressId();
+                    String address = networkAddressInventoryCache.get(networkAddressId).getName();
+                    int serviceId = serviceInventoryCache.getServiceId(networkAddressId);
+                    ServiceMapping serviceMapping = new ServiceMapping();
+                    serviceMapping.setServiceId(serviceId);
+
+                    if (config.getStaticGatewaysConfig().isAddressConfiguredAsGateway(address)) {
+                        serviceMapping.setMappingServiceId(Const.NONE);
+                        serviceMapping.setForceUpdate(true);
+                    } else {
                         serviceMapping.setMappingServiceId(segmentCoreInfo.getServiceId());
-                        serviceMappings.add(serviceMapping);
+                        serviceMapping.setForceUpdate(false);
```

Comment:
Keep `serviceMappings` as normal update, and add a list reset mappings for this case.